File: sdk/python/apibrew/configloader.py
import os
import yaml

from apibrew.config import Config


class ConfigLoader:
    @staticmethod
    def load():
        home = os.path.expanduser("~")
        apbr_config_file = home + '/.apbr/config'
        config_data = open(apbr_config_file, 'r').read()
        return Config(**yaml.load(config_data, yaml.FullLoader))

File: sdk/nodejs/src/config-loader.ts
import {Config} from "./config";
import * as yaml from "js-yaml";

let moduleNames = [];

export class ConfigLoader {
    public static load(): Config {
        const os = require('os')
        const fs = require('fs')
        const home = os.homedir();

        const apbrConfigFile = home + '/.apbr/config';

        const configData = fs.readFileSync(apbrConfigFile, 'utf8');

        return yaml.load(configData) as any;
    }

    public static loadServerConfig(serverName: string) {
        const config = ConfigLoader.load();

        if (!serverName) {
            serverName = config.defaultServer;
        }

        const serverConfig = config.servers.find((item: any) => item.name == serverName);

        if (!serverConfig) {
            throw new Error("Server not found: " + serverName);
        }

        return serverConfig
    }
}
